content/articles/how-to-analyze-machine-learning-models-using-shap/index.md

```diff
@@ -0,0 +1,566 @@
+- [Dataset scaling](#dataset-scaling)
+- [Splitting our dataset](#splitting-our-dataset)
+- [Building the model](#building-the-model)
+- [Calculate the accuracy score](#calculate-the-accuracy-score)
+- [Testing our model](#testing-our-model)
+- [Making a single prediction](#making-a-single-prediction)
+- [Getting started with SHAP](#getting-started-with-shap)
+- [Initialize KernelExplainer](#initialize-kernelexplainer)
+- [Creating shapley values](#creating-shapley-values)
+- [Force plot diagram](#force-plot-diagram)
+- [Analyzing the force plot](#analyzing-the-force-plot)
+- [Plotting a summary plot](#plotting-a-summary-plot)
+- [Conclusion](#conclusion)
+- [References](#references)
+
+### Prerequisites
+
+To understand this tutorial, a reader should:
+
+- Install [Python](https://www.python.org/) in their machine.
+- Understand [Python programming.](/engineering-education/python-projects-for-beginners/)
+- Know how to build [machine learning models](engineering-education/house-price-prediction/) using [Scikit-learn](https://scikit-learn.org/stable/)
+- Know how to use [Google colab notebooks](https://colab.research.google.com) for projects.
+- Know how to use [Pandas](https://pandas.pydata.org/) and [Numpy](https://numpy.org/) for machine learning.
+
+### Introduction
+
+Explainable AI describes the general structure of the machine learning model. It analyzes how the model features and attributes impact the model results.
+It also checks the potential biases in the model results.
+
+#### Importance of Explainable AI
+
+The reasons are as follows.
+
+- Understand the model internal functionalities and decision-making processes
+  Through Explainable AI, we can understand the internal functionalities of the model. It gives us a deeper understanding of how these models made the decisions. It enables the user to understand the criteria used to make these decisions.
+- Reduce model bias
+  When analyzing the machine learning model, we select the best features to build the model. This reduces the bias made by the model during prediction.
+- Improve the model performance
+  It improves model performance by selecting the best features to make predictions.
+- Helps in model debugging
+  Through model debugging, it removes bugs and errors in the model.
+- Identify the wrong prediction.
+  Explainable AI determines the wrong predictions made by a model. The model which makes the wrong prediction will not be deployed to production.
+
+Let's start working with our dataset.
+
+### Modelling dataset
+
+The dataset collected contains information for both diabetic and non-diabetic patients. This dataset will be used to train our model.
+
+Our model will learn from this dataset, understand patterns. The model will improve from experience. It eventually uses this information to make predictions.
+
+Let's have a look at this dataset.
+
+![Dataset image](/engineering-education/how-to-analyze-machine-learning-models-using-shap/dataset-image.jpg)
```

Comment:
Provide the reader with a link to the dataset, to ensure they follow without a hassle and test your results for themselves